ADR/00xx-component-references.md

```diff
@@ -0,0 +1,155 @@
+* [build-service] will propagate the digest reference as a PR to the bundle image, which happens to be the same git repository as the controller Component.
+  * The user submitted their controller image update on the `new-feature` branch.
+  * [build-service] could do one of the following:
+    * ❌ Push its commits with the new image digest references to the same feature branch.
+      * Will [build-service] -- acting as the TAP github App -- have rights to push to that branch? Probably not.
+    * 🤔 Submit its commits as a suggestion on the original PR, which the user can accept or not.
+      * This would be “odd” for build service, but maybe nice from the user’s point of view. Just one PR to work with.
+    * ❌ Push its commits with the new image digest references to a new feature branch based on the original feature branch, called `new-feature-build-service`
+      * This could be nice, but the PR will trigger new builds of the original image since it contains code changes for the controller (here, I am assuming that we can limit builds in PaC based on contextDirectories for multi-Component repos; we don’t have this today, but assume that we do).
+    * ✅ Push its commits with the new image digest references to a new feature branch that is not based on the original feature branch, called `new-feature-from-build-service`
+      * This works like it does in most other cases. Let the user merge the original PR, and only after that will [integration-service] update the checks for `new-feature-from-build-service` to say “okay to merge now!”
+
+## Open Questions
+
+* To give the user some control, should we make [build-service] respect [renovatebot
+  configuration files](https://docs.renovatebot.com/configuration-options/)? Does that still make
+  sense if renovatebot is not being run on a cron basis? (If the answer to the original question is
+  yes, then that may inform whether how we implement the new functionality in [build-service].
+* Since we don't have exclusive control over merging changes (or really, any control at all
+  exclusive or not), we can't prevent the user from merging things out of order. This means they can
+  get themselves into trouble. We try to provide feedback via github checks to help prevent this;
+  but we can't stop it. How does the user recover from this situation? Are there any side-effects?
+
+## Consequences
+
+* The basics of AppStudio builds are unchanged with this design (an alternative design that we considered involved changing the way pull-request build pipelines were defined and processed by PaC). This means less change spidering out to other AppStudio systems to support this change.
+* The user can pin their digest references in git. AppStudio will automate maintaining them. No magical resolution of tags in the buildsystem at build time, or worse at runtime.
+* The user is going to get more PRs on their repo. Maybe too many PRs.
```

Comment:
More PRs does not look good with the current state of RHTAP, if they trigger PLR that will run in parallel.

The current quotas on PVC is limiting how many PLRs can be run at any one time (12 as of today). There are already users being impacted and complaining about this.

Reply:
Ack. Will add this text to the consequences section.